### PR TITLE
feat: Turbine-core implementation for Turbine-PY 

### PR DESCRIPTION
**Pull request by anna-cross**
# Description

As part of Turbine core project, we are making following changes to the turbine-py library: 

- [x] Update the public library functions to map to the associated gRPC calls. - https://github.com/meroxa/turbine-py/issues/208
- [x] [Expose record function from runner - https://github.com/meroxa/turbine-py/issues/209
- [x]  Implement GRPC client init - https://github.com/meroxa/turbine-py/issues/207 

The implementation plan - 

https://www.notion.so/Turbine-Core-Migration-Implementation-Plan-5d2e805db7bc47b3b53513b3442f12e3

## Type of change

Please delete options that are not relevant.

- [x] Refactor

# How Has This Been Tested?

To test the following code - 

1.  Modify setup.config to point to pkg/ code (turbine-core implementation) instead of src/ code (old code) 
2. Run - `pip install -U turbine-py/ ` on your turbine-py
3. You should be able to now execute new core commands, such as `run` 
4. You can grab latests cli changes for turbine-py that implement run command for core - https://github.com/meroxa/cli/pull/715
5. Create a data app and run "meroxa app run ." 

- [ ] Unit Tests
- [x] Manual Tests
- [ ] Deployed to staging

### CODE REVIEW

File: pkg/turbine/cli.py [hunk 1]
```diff
@@ -0,0 +1,65 @@
+import argparse
+import asyncio
+
+from .src.turbine_app import TurbineClient
+
+# from importlib.metadata import distribution
```
Comment (ericcheatham) [inline pkg/turbine/cli.py:6]: Found some unused imports throughout the PR. Decided to leave them in leu of making decisions without the complete context.

File: pkg/turbine/src/function_app/record.py [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+from ..turbine_app import Record
+
+# from ..turbine_app import RecordList
+
+# import logging
+# from collections import UserList
+#
+# from ..turbine_app import Record
+# from ..turbine_app import RecordList
+# KTLO: https://github.com/meroxa/turbine-py/issues/151
```
Comment (ericcheatham) [inline pkg/turbine/src/function_app/record.py:12]: Is this still valid for this context?

File: pyproject.toml [hunk 3]
```diff
@@ -9,3 +9,12 @@ asyncio_mode="auto"
 testpaths = [
     "tests",
 ]
+
+[tool.black]
```
Comment (ericcheatham) [inline pyproject.toml:13]: Added this here to exclude the generated file for the proto server stuff. is there are additional files we should ignore for linting purposes please add them here.

**Comment by ericcheatham** [comment]
@anna-cross I would recommend using the `make install-hooks` [make command](https://github.com/meroxa/turbine-py/blob/main/Makefile#L15) to install the precommit hooks to your local virtual environment. This will take care of things like autoformatting and the like.

File: pyproject.toml (hunk 3 above)
Comment (anna-cross) [reply pyproject.toml:13]: I think just proto for now! :)

**Review by ericcheatham — APPROVED** [review]

**Pull request merged**